airflow/ui/src/pages/Run/Header.tsx

```diff
@@ -74,6 +74,10 @@ export const Header = ({
       { label: "Start", value: <Time datetime={dagRun.start_date} /> },
       { label: "End", value: <Time datetime={dagRun.end_date} /> },
       { label: "Duration", value: `${getDuration(dagRun.start_date, dagRun.end_date)}s` },
+      {
+        label: "Dag Versions at time of run",
+        value: dagRun.dag_versions.map((dagVersion) => `v${dagVersion.version_number}`).join(", "),
```

Comment:
Take a look at `DagTags.tsx`. I think we want to limit the number we display at once and then provide a tooltip to show all

Reply:
Done, looks like this, I'll check also to deduplicate the component. It doens't have to be DagTags and `DagVersions` most of it is the same

Reply:
(That's fake duplicated front-end data just for the purpose of testing)
![Screenshot 2025-02-26 at 18 14 03](https://github.com/user-attachments/assets/bb32d376-8d6f-410f-86ba-af0c949a92d2)